File: fronted/src/api/performance.ts
import { del, get, post } from './http'

export interface PerformanceMetric {
  name: string
  value: number
  unit: string
  threshold?: number | null
}

export interface PerformanceFinding {
  title: string
  severity: string
  description: string
  suggestion: string
}

export interface PerformanceAnalysis {
  score: number
  summary: string
  findings: PerformanceFinding[]
  trends: string[]
}

export interface PerformanceConfigs {
  name: string
  source: string
  scenario: string
  raw_text: string
  metrics: PerformanceMetric[]
  analysis?: PerformanceAnalysis | null
}

export interface PerformanceRecord {
  id: number
  project_id?: number | null
  configs: PerformanceConfigs
  created_at?: string | null
  updated_at?: string | null
}

export interface PerformanceAnalyzeRequest {
  project_id?: number | null
  name: string
  scenario: string
  raw_text: string
  metrics?: PerformanceMetric[]
}

export interface PerformanceAnalyzeResponse {
  record: PerformanceRecord
  analysis: PerformanceAnalysis
  elapsed_ms: number
}

export function analyzePerformance(data: PerformanceAnalyzeRequest): Promise<PerformanceAnalyzeResponse> {
  return post<PerformanceAnalyzeResponse>('/api/v1/performance/analyze', data)
}

export function getPerformanceRecords(projectId?: number): Promise<PerformanceRecord[]> {
  return get<PerformanceRecord[]>('/api/v1/performance/', projectId ? { project_id: projectId } : undefined)
}

export function getPerformanceRecord(recordId: number): Promise<PerformanceRecord> {
  return get<PerformanceRecord>(`/api/v1/performance/${recordId}`)
}

export function deletePerformanceRecord(recordId: number): Promise<PerformanceRecord> {
  return del<PerformanceRecord>(`/api/v1/performance/${recordId}`)
}

File: backend/app/schemas/performance.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, ConfigDict, Field


class PerformanceMetric(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    value: float
    unit: str = Field(default="", max_length=32)
    threshold: Optional[float] = None


class PerformanceFinding(BaseModel):
    title: str
    severity: str = "medium"
    description: str
    suggestion: str


class PerformanceAnalysis(BaseModel):
    score: int = Field(ge=0, le=100)
    summary: str
    findings: list[PerformanceFinding] = Field(default_factory=list)
    trends: list[str] = Field(default_factory=list)


class PerformanceConfigs(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    source: str = Field(default="manual", max_length=64)
    scenario: str = Field(default="", max_length=255)
    raw_text: str = Field(default="", max_length=8000)
    metrics: list[PerformanceMetric] = Field(default_factory=list)
    analysis: Optional[PerformanceAnalysis] = None


class PerformanceCreate(BaseModel):
    project_id: Optional[int] = None
    configs: PerformanceConfigs


class PerformanceUpdate(BaseModel):
    project_id: Optional[int] = None
    configs: Optional[PerformanceConfigs] = None


class PerformanceOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: Optional[int] = None
    configs: PerformanceConfigs
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class PerformanceAnalyzeRequest(BaseModel):
    project_id: Optional[int] = None
    name: str = Field(min_length=1, max_length=255)
    scenario: str = Field(default="", max_length=255)
    raw_text: str = Field(default="", max_length=8000)
    metrics: list[PerformanceMetric] = Field(default_factory=list)


class PerformanceAnalyzeResponse(BaseModel):
    record: PerformanceOut
    analysis: PerformanceAnalysis
    elapsed_ms: int
